### PR TITLE
CDT-335 read header from schema

### PR DESCRIPTION
**Pull request by g9-mimizhao**
- Adds optional config `infer_schema` that will force the tap to read the header from a provided catalog when no header is provided in a file

### CODE REVIEW

File: tap_s3_csv/sync.py [hunk 1]
```diff
@@ -136,7 +144,25 @@ def sync_table_file(config: Dict, s3_path: str, table_spec: Dict, stream: Dict)
...
     if s3.SDC_SOURCE_LINENO_COLUMN in reduced_table_spec["key_properties"]:
         reduced_table_spec["key_properties"].remove(s3.SDC_SOURCE_LINENO_COLUMN)
-    iterator = get_row_iterator(s3_file_stream, reduced_table_spec)  # pylint:disable=protected-access
+
+    custom_headers = False
+    fields = None
+
+
+    # fetches schema from catalog.
+    if config.get('infer_schema', None) and config['infer_schema'].lower() == "false":
```
Comment (prratek) [inline tap_s3_csv/sync.py]: if `infer_schema` is null then `config.get('infer_schema', None)` will evaluate to `None`. won't `None.lower()` fail with a `None type has no attribute lower()` or something like that?. if you change to something like this, the second part of the if won't ever evaluate if the first part is false.
```suggestion
    if 'infer_schema' in config and config['infer_schema'].lower() == "false":
```

Comment (prratek) [reply tap_s3_csv/sync.py]: try it out thought. i didn't test my suggestion

**Review by prratek — CHANGES_REQUESTED** [review]
had one suggestion, otherwise looks good

File: tap_s3_csv/sync.py (hunk 1 above)
Comment (g9-mimizhao) [reply tap_s3_csv/sync.py]: Tried it out. Works on PS and Moat so I'm gonna say its good

**Pull request merged**